### ISSUE TITLE
`hs.window.animationDuration` regression?

### ISSUE THREAD

**Issue by luckman212** [Hammerspoon/hammerspoon#3035]
- hs 0.9.93 (6148)
- macOS 12.1
- M1 MacMini
 
All of a sudden my window moves are ignoring the `hs.window.animationDuration = 0.05` that I've been using for ages. Nothing has changed in my configs. I've rebooted, and also tried disabling/re-enabling Accessibility permissions. Not sure what happened... anything I can check or try?

Thanks

**Comment by luckman212** [comment]
Just want to add this: setting e.g. `hs.window.animationDuration = 10` does cause the windows to move verrrry slowly, so the function is not completely ignoring the setting. Just that `0` or short values like `0.05` aren't working as expected.

**Comment by cmsj** [comment]
If you set it to 0, do the animations happen instantly?

**Comment by luckman212** [comment]
No, they still take around 0.75~1s.

**Comment by luckman212** [comment]
**update 3:** gets weirder. I thought I'd try creating a fresh user account to test with. Lo and behold, no problem there. Windows move / snap instantly. So there must be a per-user setting somewhere that's gotten out of whack in my main account. Something Accessibility-related I'm guessing. But no idea what that could be...

**Comment by jameslittle230** [comment]
I'm also experiencing this!

**Comment by luckman212** [comment]
Glad I'm not the only one. I can't figure this one out. Trying to compile a debug build from the source but running into some problems with the build script. Will post separately on that.

**Comment by jameslittle230** [comment]
I don't think it's a macOS 12.1 regression -- I'm on 12.0.1.

**Comment by luckman212** [comment]
Agreed. As I pointed out in my earler comment: when I created a brand new test user account and ran HS inside of it, the windows were moving/resizing normally. So it's some user setting somewhere. I monkeyed around for a few hours trying to find out which one, and failed.

**Comment by luckman212** [comment]
@jameslittle230 Happy holidays🎄

Did you figure this out by any chance? It has me stumped. I've tried tossing various `.plist` files, clearing OS caches, rebuilding the app from source—even reinstalled the OS (!) but so far nothing has worked. Works fine from a fresh user account, so it's something in `$HOME` ... but no idea what!

I checked everything reported by `sudo lsof -p $(pgrep -f 'Hammerspoon$') | grep -E 'REG.*/(Users|private)' | awk '{print $9}'` but it doesn't appear to be anything in there.

**Comment by luckman212** [comment]
Another data point:

Using the OS-native _Move Window to Left/Right Side of Screen_ function is instantaneous.

![image](https://user-images.githubusercontent.com/1992842/147707620-53d92efe-a4aa-445c-a6b9-c5f597e40e8b.png)

I collected various "window animation" related settings from around the web and assembled them into [this script](https://gist.github.com/luckman212/312216ad9b9e5d6985dbea47bb70b575) for testing. But, even when turning all those tweaks on (or deleting them), macOS did not give 2 "ducks"🦆 and continued to disobey.

Not sure what voodoo is at work here.

**Comment by jameslittle230** [comment]
@luckman212 - you were working on compiling the project, right? Did you ever figure out if this is a hammerspoon project regression? In other words, if you compile a previous version, does it still exhibit this smooth resize behavior?

**Comment by luckman212** [comment]
I haven't compiled an old version, but I did compile the latest code (no change) as well as try the 0.9.91 and 0.9.92 releases. No luck. Again- starting from a fresh user account it seems to work fine. 

I just need to narrow down which of these 9,521 prefs is causing it. 😳

```shell
$ find ~ -type f -name "*.plist" | wc -l
    9521
```

**Comment by luckman212** [comment]
@jameslittle230 and @cmsj Happy new year! 🎉 

I think I may have found the root cause (and a temporary workaround) for the slow window animations. It seems to be related to the following setting from Accessibility → Zoom → Advanced → Follow Focus → **Follow keyboard focus**.

On my system, anything except **Never** results in the animations slowing down:

<img src="https://user-images.githubusercontent.com/1992842/147885052-6364ebb1-95a9-4e1b-be4e-f936f472c7ff.png" width="550">

You can try disabling that via the GUI or using this Terminal command:
```shell
defaults write com.apple.universalaccess closeViewZoomFocusFollowModeKey -int 0
```

_**N.B.** you must log out and back in for that change to take effect._

Please give it try & let me know if your animations are snappy again.

**Comment by jameslittle230** [comment]
My Follow Keyboard Focus was set to Never just now when I went to check, and I'm pretty sure it's been set to Never this whole time.

HOWEVER, I just reloaded my HS config for an unrelated reason and my window movements are immediate again! So I guess this resolved itself for me on its own? Very mysterious.

**Comment by luckman212** [comment]
@jameslittle230 Interesting! Another way I found that (temporarily) slows down the animations is this:

1. open System Preferences → Keyboard
2. switch to the **Input Sources** pane
3. enable the **Show Input menu in menu bar** checkbox
4. from the Keyboard menubar item, choose **Show Keyboard Viewer**

When I do this, my animations get slow again until I log out and back in. Again, not sure what voodoo is going on deep in the Accessibility subsystem to cause this but—there it is. Perhaps this was something you did before?

**Comment by luckman212** [comment]
@cmsj Curious if you have any thoughts on this one? Should I file a bug with Apple or do you think this is somehow fixable in HS?

**Comment by MoonSoul88** [comment]
> @jameslittle230
solved this way

 Accessibility → Zoom

if check this option "animation delay" Occurs immediately.
option check out and Reboot!

<img width="500" alt="스크린샷 2022-08-21 오전 9 22 16" src="https://user-images.githubusercontent.com/6717388/185770333-ee6eb968-30fd-40e6-9a0f-a49ab1f982f5.png">

**Comment by MoonSoul88** [comment]
solved this way

 Accessibility → Zoom

if check this option "animation delay" Occurs immediately.
option check out and Reboot!

<img width="500" alt="스크린샷 2022-08-21 오전 9 22 16" src="https://user-images.githubusercontent.com/6717388/185770568-b232e683-bd3b-42bd-9d8a-b19b2890f68a.png">

**Comment by Rhys-T** [comment]
This sounds like it's probably a result of the `AXEnhancedUserInterface` property getting enabled. `AXEnhancedUserInterface` is an undocumented property that was originally set to indicate that VoiceOver was running, but some third-party apps - including Grammarly and 1Password - also set it to true, to make sure that they can see inside Chrome and related apps. Other macOS accessibility features, like the Keyboard Viewer, also seem to set it.

Unfortunately, it has some weird side effects, like adding animations when you try to move/resize anything through the Accessibility API - and canceling those animations halfway through if you try to do a move immediately followed by a resize. See https://github.com/Hammerspoon/hammerspoon/issues/3224#issuecomment-1294359070 and replies for more info and workarounds.
